src/solvers/interior_point_Newton.jl

```diff
@@ -0,0 +1,382 @@
+    interior_point_Newton(M, ConstrainedManifoldObjective, p; kwargs...)
+
+perform the interior point Newton method following [LaiYoshise:2024](@cite).
+
+In order to solve the constrained problem
+
+```math
+\begin{aligned}
+\min_{p ∈\mathcal{M}} &f(p)\\
+\text{subject to } &g_i(p)\leq 0 \quad \text{ for } i= 1, …, m,\\
+\quad &h_j(p)=0 \quad \text{ for } j=1,…,n,
+\end{aligned}
+```
+
+This algorithms iteratively solves the linear system based on extending the KKT system
+by a slack variable `s`.
+
+```math
+\operatorname{J} F(p, μ, λ, s)[X, Y, Z, W] = -F(p, μ, λ, s),
+\text{ where }
+X ∈ T_p\mathcal M, Y,W ∈ ℝ^m, Z ∈ ℝ^n,
+```
+
+see [`CondensedKKTVectorFieldJacobian`](@ref) and [`CondensedKKTVectorField`](@ref), respectively,
+for the reduced form, this is usually solved in.
+From the resulting `X` and `Z` in the reeuced form, the other two, ``Y``, ``W``, are then computed.
+
+From the gradient ``(X,y,Z,W)`` at the current iterate ``(p, μ, λ, s)``,
+a line search is performed using the [`KKTVectorFieldNormSq`](@ref) norm of the KKT vector field (squared)
+and its gradient [`KKTVectorFieldNormSqGradient`](@ref) together with the [`InteriorPointCentralityCondition`](@ref).
+
+Note that since the vector field ``F`` includes the gradients of the constraint
+functions ``g,h`, its gradient or Jacobian requires the Hessians of the constraints.
+
+For that seach direction a line search is performed, that additionally ensures that
+the constraints are further fulfilled.
+
+# Input
+
+* `M`:      a manifold ``\mathcal M``
+* `f`:      a cost function ``f : \mathcal M → ℝ`` to minimize
+* `grad_f`: the gradient ``\operatorname{grad} f : \mathcal M → T \mathcal M`` of ``f``
+* `Hess_f`: the Hessian ``\operatorname{Hess}f(p): T_p\mathcal M → T_p\mathcal M``, ``X ↦ \operatorname{Hess}f(p)[X] = ∇_X\operatorname{grad}f(p)``
+* `p=`[`rand`](@extref Base.rand-Tuple{AbstractManifold})`(M)`: an initial value ``p  ∈  \mathcal M``
+
+or a [`ConstrainedManifoldObjective`](@ref) `cmo` containing `f`, `grad_f`, `Hess_f`, and the constraints
+
+# Keyword arguments
+
+The keyword arguments related to the constraints (the first eleven) are ignored if you
+pass a [`ConstrainedManifoldObjective`](@ref) `cmo`
+
+* `centrality_condition=missing`; an additional condition when to accept a step size.
+  This can be used to ensure that the resulting iterate is still an interior point if you provide a check `(N,q) -> true/false`,
+  where `N` is the manifold of the `step_problem`.
+* `equality_constraints=nothing`: the number ``n`` of equality constraints.
+* `evaluation=`[`AllocatingEvaluation`](@ref)`()`:
+  specify whether the functions that return an array, for example a point or a tangent vector, work by allocating its result ([`AllocatingEvaluation`](@ref)) or whether they modify their input argument to return the result therein ([`InplaceEvaluation`](@ref)). Since usually the first argument is the manifold, the modified argument is the second."
```

Comment:
```suggestion
  specify whether the functions that return an array, for example a point or a tangent vector, work by allocating its result ([`AllocatingEvaluation`](@ref)) or whether they modify their input argument to return the result therein ([`InplaceEvaluation`](@ref)). Since usually the first argument is the manifold, the modified argument is the second.
```